src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -0,0 +1,360 @@
+            .iter()
+            .cloned()
+            .collect_vec();
+
+        debug_assert!(!member_table_ids_1.is_empty());
+        debug_assert!(!member_table_ids_2.is_empty());
+        assert!(member_table_ids_1.is_sorted());
+        assert!(member_table_ids_2.is_sorted());
+
+        // Make sure `member_table_ids_1` is smaller than `member_table_ids_2`
+        let (left_group_id, right_group_id) =
+            if member_table_ids_1.first().unwrap() < member_table_ids_2.first().unwrap() {
+                (group_1, group_2)
+            } else {
+                std::mem::swap(&mut member_table_ids_1, &mut member_table_ids_2);
+                (group_2, group_1)
+            };
+
+        // We can only merge two groups with non-overlapping member table ids
+        if member_table_ids_1.last().unwrap() >= member_table_ids_2.first().unwrap() {
+            return Err(Error::CompactionGroup(format!(
+                "invalid merge group_1 {} group_2 {}",
+                left_group_id, right_group_id
+            )));
+        }
+
+        let combined_member_table_ids = member_table_ids_1
+            .iter()
+            .chain(member_table_ids_2.iter())
+            .collect_vec();
+        assert!(combined_member_table_ids.is_sorted());
+
+        // check duplicated sst_id
+        let mut sst_id_set = HashSet::new();
+        for sst_id in versioning
+            .current_version
+            .get_sst_ids_by_group_id(left_group_id)
+            .chain(
+                versioning
+                    .current_version
+                    .get_sst_ids_by_group_id(right_group_id),
+            )
+        {
+            if !sst_id_set.insert(sst_id) {
+                return Err(Error::CompactionGroup(format!(
+                    "invalid merge group_1 {} group_2 {} duplicated sst_id {}",
+                    left_group_id, right_group_id, sst_id
+                )));
+            }
+        }
+
+        // TODO(li0k): remove this check (Since the current split_sst does not change key_range, this check can not be removed, otherwise concate will fail.)
+        // check branched sst on non-overlap level
+        {
+            let left_levels = versioning
+                .current_version
+                .get_compaction_group_levels(group_1);
+
+            let right_levels = versioning
+                .current_version
+                .get_compaction_group_levels(group_2);
+
+            // we can not check the l0 sub level, because the sub level id will be rewritten when merge
+            // This check will ensure that other non-overlapping level ssts can be concat and that the key_range is correct.
+            let max_level = std::cmp::max(left_levels.levels.len(), right_levels.levels.len());
+            for level_idx in 0..max_level {
+                let left_level = left_levels.levels.get(level_idx).unwrap();
+                let right_level = right_levels.levels.get(level_idx).unwrap();
```

Comment:
It is confusing that `levels.get(level_idx)` is getting level `level_idx + 1`. At the first glance, I thought we are checking level 0 as well here. How about using the [`get_level`](https://github.com/risingwavelabs/risingwave/blob/4af2cd36dcdd3fcdd850a0c84a3bbce47e1c50ff/src/storage/hummock_sdk/src/level.rs#L214) here?